### ISSUE TITLE
Please supply to the build script the number of CPUs that build will be using

### ISSUE THREAD

**Issue by yurivict** [bazelbuild/bazel#2048]
I build bazel with ```./compile.sh``` command. It used many CPUs, and hangs on FreeBSD due to this bug https://github.com/bazelbuild/bazel/issues/1250.

There needs to be the variable passed to ```compile.sh``` that will tell it to use this number of CPUs. The ```--jobs {JOBS}``` should be exposed as the build argument.

Besides the workaround for the above-mentioned bug, it is also good to be able to pass the number of jobs to the build from the port variable that we have generally defined.

**Comment by yurivict** [comment]
Actually, adding ```--jobs {JOBS} ``` to ```scripts/bootstrap/compile.sh``` doesn't work. Build still runs on 4 CPUs for me and hangs. Is there any workaround for this? I can't finish the FreeBSD port because of this.

**Comment by petemounce** [comment]
I think the [`--local_resources`](https://bazel.build/versions/master/docs/bazel-user-manual.html) is what you want here:
> --local_resources availableRAM,availableCPU,availableIO
> 
> This option, which takes three comma-separated floating point arguments, specifies the amount of local resources that Bazel can take into consideration when scheduling build and test activities. Option expects amount of available RAM (in MB), number of CPU cores (with 1.0 representing single full core) and workstation I/O capability (with 1.0 representing average workstation). By default Bazel will estimate amount of RAM and number of CPU cores directly from system configuration and will assume 1.0 I/O resource.
> 
> If this option is used, Bazel will ignore --ram_utilization_factor.

Beware, you need to supply all 3 arguments, I think. I don't think `--local_resources=,4,` will work (for `don't care,4,don't care`, sadly.

**Comment by damienmg** [comment]
`export EXTRA_BAZEL_ARGS='--jobs 1'`

**Comment by yurivict** [comment]
I just tried to supply ```EXTRA_BAZEL_ARGS='--jobs 1'``` environment variable to ```compile.sh```, and it didn't make a difference.

As for ```--local_resources```, it looks like a ```bazel``` argument, not a build argument.

How do I make ```compile.sh``` to run on only 1 CPU?

**Comment by damienmg** [comment]
I got it wrong, it is "BAZEL_EXTRA_ARGS"

**Comment by yurivict** [comment]
```BAZEL_EXTRA_ARGS='--jobs 1'``` made some  difference, but multiple processes are still run in parallel, see below, and it still hangs.



```
 | |                 \-+- 19720 root /usr/local/bin/bash ./compile.sh
 | |                   \-+- 20330 root /usr/local/openjdk8/bin/java -XX:+HeapDumpOnOutOfMemoryError -Xverify:none -Dfile.encoding=ISO-8859-1 -XX:HeapDumpPath=/tmp/bazel_11J8su2q -Djava.util.logging.config.file=/tmp/bazel_11J8su2q/javalog.properties -Dio.bazel.EnableJni=0 -jar /tmp/bazel_11J8su2q/archive/libblaze.jar --batch --install_base=/tmp/bazel_11J8su2q/archive --output_base=/tmp/bazel_11J8su2q/out --install_md5= --workspace_directory=/usr/ports/devel/bazel/work/bazel-0.4.0 --nofatal_event_bus_exceptions build --ignore_unsupported_sandboxing --startup_time=329 --extract_data_time=523 --rc_source=/dev/null --isatty=1 --ignore_client_env --client_cwd=/usr/ports/devel/bazel/work/bazel-0.4.0 --java_toolchain=//src/java_tools/buildjar:bootstrap_toolchain --host_java_toolchain=//src/java_tools/buildjar:bootstrap_toolchain --strategy=Javac=worker --worker_quit_after_build --ignore_unsupported_sandboxing --compilation_mode=opt --verbose_failures --javacopt=-g -source 1.8 -target 1.8 --stamp --embed_label 0.4.0- (@non-git) src:bazel
 | |                     |--- 22868 root /tmp/bazel_11J8su2q/out/execroot/bazel-0.4.0/external/local_jdk/bin/java -Xbootclasspath/p:third_party/java/jdk/langtools/javac.jar -XX:+TieredCompilation -XX:TieredStopAtLevel=1 -jar bazel-out/host/bin/src/java_tools/buildjar/java/com/google/devtools/build/buildjar/bootstrap_deploy.jar --persistent_worker
 | |                     |--- 22869 root /tmp/bazel_11J8su2q/out/execroot/bazel-0.4.0/external/local_jdk/bin/java -Xbootclasspath/p:third_party/java/jdk/langtools/javac.jar -XX:+TieredCompilation -XX:TieredStopAtLevel=1 -jar bazel-out/host/bin/src/java_tools/buildjar/java/com/google/devtools/build/buildjar/bootstrap_deploy.jar --persistent_worker
 | |                     |--- 22870 root /tmp/bazel_11J8su2q/out/execroot/bazel-0.4.0/external/local_jdk/bin/java -Xbootclasspath/p:third_party/java/jdk/langtools/javac.jar -XX:+TieredCompilation -XX:TieredStopAtLevel=1 -jar bazel-out/host/bin/src/java_tools/buildjar/java/com/google/devtools/build/buildjar/bootstrap_deploy.jar --persistent_worker
 | |                     \--- 22871 root /tmp/bazel_11J8su2q/out/execroot/bazel-0.4.0/external/local_jdk/bin/java -Xbootclasspath/p:third_party/java/jdk/langtools/javac.jar -XX:+TieredCompilation -XX:TieredStopAtLevel=1 -jar bazel-out/host/bin/src/java_tools/buildjar/java/com/google/devtools/build/buildjar/bootstrap_deploy.jar --persistent_worker
```

**Comment by aehlig** [comment]
Tagged as feature request, as the underlying hang is already tracked in #1250.  Will look into it more closely, once distribution-artifact patch series (ending in https://bazel-review.googlesource.com/#/c/6831/) is in so I can test bootstrapping on FreeBSD without using the linux ABI.

**Comment by aehlig** [comment]
Looking at the process tree again, the additional process really are the persistent workers. The workaround for FreeBSD is known (https://svnweb.freebsd.org/ports/head/devel/bazel/files/patch-scripts_bootstrap_bootstrap.sh?revision=436446&view=markup) even though it requires a patch to `compile.sh`. As the issue as it stands (restricting the number of jobs) can be solved by `BAZEL_EXTRA_ARGS='--jobs 1'`, what remains is essentially a duplicate of #1250.

Feel free to file a feature request to make the `Javac` strategy in `compile.sh` settable by environment variables, if this feature is still needed.
